api/db/changelog/db.changelog-225-add-user-discovery-source.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+  xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+  <changeSet author="Peter-Lavigne" id="changelog-225-add-user-discovery-source">
+    <addColumn tableName="user">
+      <column name="user_general_discovery_source_other_text" type="varchar(255)"/>
+      <column name="user_partner_discovery_source_other_text" type="varchar(255)"/>
+    </addColumn>
+
+    <createTable tableName="user_general_discovery_source">
+      <column name="user_id" type="bigint">
+        <constraints
+                nullable="false"
+                foreignKeyName="fk_user_general_discovery_source_user"
+                references="user(user_id)"
+                deleteCascade="true"
+        />
+      </column>
+
+      <column name="source" type="ENUM('RESEARCH_ALL_OF_US_WEBSITE','SOCIAL_MEDIA','JOURNAL_OR_NEWS_ARTICLE','ACTIVITY_PRESENTATION_OR_EVENT','FRIENDS_OR_COLLEAGUES','OTHER_WEBSITE','OTHER')">
+        <constraints nullable="false"/>
+      </column>
+    </createTable>
+
+    <createTable tableName="user_partner_discovery_source">
+      <column name="user_id" type="bigint">
+        <constraints
+                nullable="false"
+                foreignKeyName="fk_user_partner_discovery_source_user"
+                references="user(user_id)"
+                deleteCascade="true"
+        />
+      </column>
+      <column name="source" type="ENUM(
+        'ALL_OF_US_RESEARCH_PROGRAM_STAFF',
+        'ALL_OF_US_EVENINGS_WITH_GENETICS_RESEARCH_PROGRAM_BAYLOR_COLLEGE_OF_MEDICINE_DEPARTMENT_OF_MOLECULAR_AND_HUMAN_GENETICS',
```

Comment:
Currently, all of the enums exactly match the question labels.

The two things I am considering changing:
- This one is quite long, and could be `ALL_OF_US_EVENINGS_WITH_GENETICS_RESEARCH_PROGRAM` instead
- Some of them end in acronyms, like `AMERICAN_ASSOCIATION_ON_HEALTH_AND_DISABILITY_AAHD`. Should the acronyms be removed?

Reply:
I renamed `ALL_OF_US_EVENINGS_WITH_GENETICS_RESEARCH_PROGRAM_BAYLOR_COLLEGE_OF_MEDICINE_DEPARTMENT_OF_MOLECULAR_AND_HUMAN_GENETICS` to `ALL_OF_US_EVENINGS_WITH_GENETICS_RESEARCH_PROGRAM`